snuba/query/expressions.py

```diff
@@ -110,10 +110,88 @@ def visit_lambda(self, exp: Lambda) -> TVisited:
         raise NotImplementedError
 
 
+class StringifyVisitor(ExpressionVisitor[str]):
+    """Visitor implementation to turn an expression into a string format
+
+    Usage:
+        # Any expression class supported by the visitor will do
+        >>> exp: Expression = Expression()
+        >>> visitor = StringifyVisitor()
+        >>> exp_str = exp.accept(visitor)
+    """
+
+    def __init__(self) -> None:
+        # keeps track of the level of the AST we are currently in, this is necessary for nice indentation
+        # before recursively going into subnodes increment this counter, decrement it after the recursion
+        # is done
+        self.__level = 0
+
+    def _get_line_prefix(self) -> str:
+        # every line in the tree needs to be indented based on the tree level
+        # to make things look pretty
+        return "\n" + "  " * self.__level
+
+    def _get_alias_str(self, exp: Expression) -> str:
+        # Every expression has an optional alias so we handle that here
+        return f" AS `{exp.alias}`" if exp.alias else ""
```

Comment:
How verbose does it become if you surround aliased expresisons with parentheses?
There are several cases where we have nested aliases:
`f(a as A, f(b AS c, d) as D) as E`
Some times wrapping the aliased expression makes it a little more manageable. But it may be overwhelming in all other cases

Reply:
Is this more readable? My gut says no but feel free to disagree

```
(f(
  (a as A), 
  (  
    f(
      (b AS c), 
      d
    ) as D
)) as E
```

Reply:
nah you are right. let's do without for now.